### PR TITLE
fixes the dep tree for stub-lite - fixes #117

### LINKED ISSUE #117: io.grpc:grpc-kotlin-stub-lite includes io.grpc:grpc-kotlin-stub
Hi! Here is my runtimeClasspath dependencies 
```
runtimeClasspath - Runtime classpath of compilation 'main' (target  (jvm)).
+--- org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.3.72
|    +--- org.jetbrains.kotlin:kotlin-stdlib:1.3.72
|    |    +--- org.jetbrains.kotlin:kotlin-stdlib-common:1.3.72
|    |    \--- org.jetbrains:annotations:13.0
|    \--- org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.3.72
|         \--- org.jetbrains.kotlin:kotlin-stdlib:1.3.72 (*)
+--- org.jetbrains.kotlinx:kotlinx-coroutines-core:1.3.7
|    +--- org.jetbrains.kotlin:kotlin-stdlib:1.3.71 -> 1.3.72 (*)
|    \--- org.jetbrains.kotlin:kotlin-stdlib-common:1.3.71 -> 1.3.72
+--- io.grpc:grpc-okhttp:1.27.2
|    +--- io.grpc:grpc-core:1.27.2
|    |    +--- io.grpc:grpc-api:1.27.2 -> 1.28.0
|    |    |    +--- io.grpc:grpc-context:1.28.0
|    |    |    +--- com.google.errorprone:error_prone_annotations:2.3.4
|    |    |    +--- com.google.code.findbugs:jsr305:3.0.2
|    |    |    +--- org.codehaus.mojo:animal-sniffer-annotations:1.18
|    |    |    \--- com.google.guava:guava:28.1-android
|    |    |         +--- com.google.guava:failureaccess:1.0.1
|    |    |         +--- com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava
|    |    |         +--- org.checkerframework:checker-compat-qual:2.5.5
|    |    |         \--- com.google.j2objc:j2objc-annotations:1.3
|    |    +--- com.google.code.gson:gson:2.8.6
|    |    +--- com.google.android:annotations:4.1.1.4
|    |    +--- com.google.errorprone:error_prone_annotations:2.3.4
|    |    \--- io.perfmark:perfmark-api:0.19.0
|    |         \--- com.google.code.findbugs:jsr305:3.0.2
|    +--- com.squareup.okio:okio:1.13.0
|    \--- com.squareup.okhttp:okhttp:2.5.0
+--- io.grpc:grpc-stub:1.27.2 -> 1.28.0
|    \--- io.grpc:grpc-api:1.28.0 (*)
+--- io.grpc:grpc-protobuf-lite:1.27.2 -> 1.28.0
|    +--- io.grpc:grpc-api:1.28.0 (*)
|    +--- com.google.protobuf:protobuf-javalite:3.11.0
|    \--- com.google.guava:guava:28.1-android (*)
\--- io.grpc:grpc-kotlin-stub-lite:0.1.3
     +--- org.jetbrains.kotlinx:kotlinx-coroutines-core:1.3.3 -> 1.3.7 (*)
     +--- io.grpc:grpc-protobuf-lite:1.28.0 (*)
     +--- io.grpc:grpc-stub:1.28.0 (*)
     +--- io.grpc:grpc-kotlin-stub:0.1.3
     |    +--- org.jetbrains.kotlinx:kotlinx-coroutines-core:1.3.3 -> 1.3.7 (*)
     |    +--- io.grpc:grpc-protobuf:1.28.0
     |    |    +--- io.grpc:grpc-api:1.28.0 (*)
     |    |    +--- com.google.protobuf:protobuf-java:3.11.0
     |    |    +--- com.google.guava:guava:28.1-android (*)
     |    |    +--- com.google.api.grpc:proto-google-common-protos:1.17.0
     |    |    \--- io.grpc:grpc-protobuf-lite:1.28.0 (*)
     |    +--- io.grpc:grpc-stub:1.28.0 (*)
     |    \--- javax.annotation:javax.annotation-api:1.2
     \--- javax.annotation:javax.annotation-api:1.2

```
As you can see `io.grpc:grpc-kotlin-stub-lite:0.1.3` includes `io.grpc:grpc-kotlin-stub:0.1.3. `
I think this is not correct.

This is my dependencies. Maybe i do something wrong?
```
    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8"
    implementation "org.jetbrains.kotlinx:kotlinx-coroutines-core:${coroutinesVersion}"

    implementation "io.grpc:grpc-okhttp:$grpcVersion"
    implementation "io.grpc:grpc-stub:$grpcVersion"
    implementation "io.grpc:grpc-protobuf-lite:$grpcVersion"
    implementation "io.grpc:grpc-kotlin-stub-lite:$grpcktVersion"

    compileOnly "javax.annotation:javax.annotation-api:1.2"
```

### PR DESCRIPTION
**Pull request by jamesward**

### CODE REVIEW

File: stub-lite/build.gradle [hunk 1]
```diff
@@ -29,10 +26,12 @@ dependencies {
     implementation "org.jetbrains.kotlinx:kotlinx-coroutines-core:${rootProject.ext.coroutinesVersion}"
 
     // Grpc and Protobuf
-    implementation "io.grpc:grpc-protobuf-lite:${rootProject.ext.grpcVersion}"
-    implementation "io.grpc:grpc-stub:${rootProject.ext.grpcVersion}"
-    implementation ("io.grpc:grpc-kotlin-stub:$version") {
-        exclude group: 'io.grpc', module: 'grpc-protobuf'
+    api "io.grpc:grpc-protobuf-lite:${rootProject.ext.grpcVersion}"
+    api "com.google.guava:guava:29.0-android"
```
Comment (jamesward) [inline stub-lite/build.gradle:30]: This feels wrong but if I don't include it, in the `examples` I get an error. To repro from this branch:

Change to the local maven and remove the dep on `com.google.guava:guava:29.0-android`:
```
diff --git a/examples/build.gradle.kts b/examples/build.gradle.kts
index 5d78d5d..e4150aa 100644
--- a/examples/build.gradle.kts
+++ b/examples/build.gradle.kts
@@ -7,7 +7,7 @@ plugins {
 
 // todo: move to subprojects, but how?
 ext["grpcVersion"] = "1.32.1"
-ext["grpcKotlinVersion"] = "0.2.0" // CURRENT_GRPC_KOTLIN_VERSION
+ext["grpcKotlinVersion"] = "0.2.1-SNAPSHOT" // CURRENT_GRPC_KOTLIN_VERSION
 ext["protobufVersion"] = "3.13.0"
 
 allprojects {
diff --git a/examples/stub-lite/build.gradle.kts b/examples/stub-lite/build.gradle.kts
index aacd6ed..3e676cb 100644
--- a/examples/stub-lite/build.gradle.kts
+++ b/examples/stub-lite/build.gradle.kts
@@ -20,13 +20,15 @@ dependencies {
     api("io.grpc:grpc-protobuf-lite:${rootProject.ext["grpcVersion"]}")
 
     // todo: I think we should be able to just include this:
-    // api("io.grpc:grpc-kotlin-stub-lite:${rootProject.ext["grpcKotlinVersion"]}")
+    api("io.grpc:grpc-kotlin-stub-lite:${rootProject.ext["grpcKotlinVersion"]}")
     // but it doesn't pull transitives correctly
 
+    /*
     api("io.grpc:grpc-stub:${rootProject.ext["grpcVersion"]}")
     api("io.grpc:grpc-kotlin-stub:${rootProject.ext["grpcKotlinVersion"]}") {
         exclude("io.grpc", "grpc-protobuf")
     }
+     */
 }
 
 java {
diff --git a/stub-lite/build.gradle b/stub-lite/build.gradle
index 3274de7..79798f2 100644
--- a/stub-lite/build.gradle
+++ b/stub-lite/build.gradle
@@ -27,7 +27,7 @@ dependencies {
 
     // Grpc and Protobuf
     api "io.grpc:grpc-protobuf-lite:${rootProject.ext.grpcVersion}"
-    api "com.google.guava:guava:29.0-android"
+    //api "com.google.guava:guava:29.0-android"
     api "io.grpc:grpc-stub:${rootProject.ext.grpcVersion}"
 
     api(project(":grpc-kotlin-stub")) {
```

Then publish `0.2.1-SNAPSHOT` locally:
```
./gradlew publishToMavenLocal
```

Now try to build the examples:
```
cd examples
./gradlew :native-client:run
```

Error is:
```
/home/jamesward/projects/grpc-kotlin/examples/stub-lite/build/generated/source/proto/main/grpc/io/grpc/examples/routeguide/RouteGuideGrpc.java:433: error: package com.google.common.util.concurrent does not exist
    public com.google.common.util.concurrent.ListenableFuture<io.grpc.examples.routeguide.Feature> getFeature(
```

Comment (jamesward) [reply stub-lite/build.gradle:30]: @saturnism Can you help with this?

**Comment by jamesward** [comment]
Once we release this, we can switch the samples to use it.

**Comment by jamesward** [comment]
Outstanding question... is `api` the right scope in `stub` and `stub-lite` for the transitive grpc deps?

**Comment by jamesward** [comment]
From what I can tell, the `api` scope is correct because the `stub` / `stub-lite` and their dependencies are needed at compile time in projects that depend on them.

File: build.gradle [hunk 2]
```diff
@@ -17,7 +17,7 @@ subprojects {
     apply plugin: 'java'
     apply plugin: 'org.jetbrains.kotlin.jvm'
     group = "io.grpc"
-    version = "0.2.0" // CURRENT_GRPC_KOTLIN_VERSION
+    version = "0.2.1-SNAPSHOT" // CURRENT_GRPC_KOTLIN_VERSION
```
[the remark this answers is not in the document]
Comment (jamesward) [reply build.gradle:20]:
```suggestion
    version = "0.2.1" // CURRENT_GRPC_KOTLIN_VERSION
```

File: build.gradle [hunk 3]
```diff
@@ -17,7 +17,7 @@ subprojects {
     apply plugin: 'java'
     apply plugin: 'org.jetbrains.kotlin.jvm'
     group = "io.grpc"
-    version = "0.2.0" // CURRENT_GRPC_KOTLIN_VERSION
+    version = "0.2.1" // CURRENT_GRPC_KOTLIN_VERSION
```
Comment (jamesward) [inline build.gradle:20]:
```suggestion
    version = "0.2.0" // CURRENT_GRPC_KOTLIN_VERSION
```

**Review by bshaffer — APPROVED** [review]

**Pull request merged**